src/Support/Eventable.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace KirschbaumDevelopment\NovaMail\Support;
+
+use Illuminate\Support\Str;
+use Symfony\Component\Finder\Finder;
+use Illuminate\Database\Eloquent\Model;
+use KirschbaumDevelopment\NovaMail\Traits\Mailable;
+
+class Eventable
+{
+    private static $defaultPaths = [
+        '',
+        'Models',
+    ];
+
+    /**
+     * Eloquent models that events should be listened on.
+     *
+     * @var array
+     */
+    private static $models = [];
+
+    /**
+     * Register all of the resource classes in the given directory.
+     *
+     * @return void
+     */
+    public static function discoverModels()
+    {
+        $namespace = app()->getNamespace();
+
+        $directories = collect(self::$defaultPaths)
+            ->merge(config('nova_mail.model_paths'))
+            ->transform(function($model) {
+                return app_path($model);
+            })
+            ->filter(function ($directory) {
+                return is_dir($directory);
+            })
+            ->toArray();
+
+        if (empty($directories)) {
+            return;
+        }
+
+        foreach ((new Finder)->in($directories)->depth(0)->files() as $resource) {
```

Comment:
@brandonferens Can I be a stickler for collection usage here?

Reply:
You can be. I don't always reach for it if it seems like unnecessary overhead without much value. However, I did some refactoring and it make more sense to move to collections. Hope you are happy! Merry Christmas!